### PR TITLE
feat: 🎸 initial LDAP authentication support

### PR DESCRIPTION
**Pull request by randallmorey**
This pr lists LDAP on the signup/login form and allows authentication via LDAP on both admin and desktop UI

[jira](https://hashicorp.atlassian.net/browse/ICU-9213)

admin ui:
Screenshot 2023-05-10 at 11 43 15 AM

Screenshot 2023-05-10 at 11 43 29 AM

Screenshot 2023-05-10 at 11 43 42 AM

desktop: 

Screenshot 2023-05-10 at 11 45 41 AM

Screenshot 2023-05-10 at 11 55 14 AM

Screenshot 2023-05-10 at 11 55 37 AM

### CODE REVIEW

File: addons/core/addon/components/form/authenticate/index.hbs [hunk 1]
```diff
@@ -9,7 +9,6 @@
   {{component
     (concat 'form/authenticate/' @model.type)
     submit=@submit
-    cancel=false
```
Comment (DhariniJeeva) [inline addons/core/addon/components/form/authenticate/index.hbs:12]: I removed this as we don't have a cancel button on the authenticate forms

File: addons/core/tests/integration/components/form/authenticate-test.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, find } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+import { setupIntl } from 'ember-intl/test-support';
+
+module('Integration | Component | form | authenticate', function (hooks) {
```
Comment (cameronperera) [inline addons/core/tests/integration/components/form/authenticate-test.js:12]: Could we add a some more tests/asserts for this form? My thoughts are with the current test, we could delete the 2 input fields and this tests would still pass. My suggestion would be to assert the inputs and form buttons are present.

**Review by cameronperera — CHANGES_REQUESTED** [review]

**Review by lisbet-alvarez — APPROVED** [review]
LGTM!

**Review by cameronperera — APPROVED** [review]
Looks great!

**Pull request merged**